crowbar_framework/app/models/dns_service.rb

```diff
@@ -188,4 +188,30 @@ def save_config_to_databag(old_role, role, server_nodes = nil)
     instance = Crowbar::DataBagConfig.instance_from_role(old_role, role)
     Crowbar::DataBagConfig.save("core", instance, @bc_name, config)
   end
+
+  # try to know if we can skip a node from running chef-client
+  def skip_unchanged_node?(node_name, old_role, new_role)
+    @logger.debug("#{@bc_name} skip_batch_for_node? entering: #{node_name}")
+
+    # if old_role is nil, then we are applying the barclamp for the first time
+    return false if old_role.nil?
+
+    # if the node changed roles, then we need to apply
+    if node_changed_roles?(node_name, old_role, new_role)
+      logger.debug("#{@bc_name} skip_batch_for_node?: not skipping #{node} (role change)")
```

Comment:
I would actually not have any "not skipping" log; that's the default behavior and can be very spammy. I think we should only log when skipping.